datafusion/sqllogictest/test_files/select.slt

```diff
@@ -550,9 +550,16 @@ select * from (select 1 a union all select 2) b order by a limit 1;
 1
 
 # select limit clause invalid
-statement error DataFusion error: Error during planning: LIMIT must not be negative
+statement error DataFusion error: Error during planning: LIMIT must be >= 0, '\-1' was provided\.
 select * from (select 1 a union all select 2) b order by a limit -1;
 
+# select limit with basic arithmetic
+query I
+select * from (select 1 a union all select 2) b order by a limit 1+1;
+----
+1
+2
+
```

Comment:
Can you also add a few error cases -- for example:

```sql
❯ select * from (values (1)) LIMIT 10*100;
Error during planning: Unsupported operator for LIMIT clause
❯ select * from (values (1)) LIMIT cast(column1 as tinyint);
Error during planning: Unexpected expression in LIMIT clause
```

Reply:
Sure, added.